src/transformStateWithClones.js

```diff
@@ -7,7 +7,41 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  let currentState = { ...state };
+  const stateHistory = [];
+
+  actions.forEach((action) => {
+    switch (action.type) {
+      case 'clear':
+        // Очищаем текущее состояние
+        currentState = {};
+        break;
+
+      case 'addProperties':
+        // Добавляем свойства из extraData
```

Comment:
Remove comment. Try to write obvious code that explains all we need without comments)